contracts/NodeDelegator.sol

```diff
@@ -291,6 +318,36 @@ contract NodeDelegator is INodeDelegator, LRTConfigRoleChecker, PausableUpgradea
         );
     }
 
+    /// @notice Undelegates all staked assets from this NodeDelegator from the
+    /// previously delegated to EigenLayer Operator.
+    /// This also forces a withdrawal so the assets will need to be claimed.
+    function undelegate() external onlyLRTManager {
+        address delegationManagerAddress = lrtConfig.getContract(LRTConstants.EIGEN_DELEGATION_MANAGER);
+        IDelegationManager delegationManager = IDelegationManager(delegationManagerAddress);
+
+        // Get the amount of strategy shares owned by this NodeDelegator contract
+        IStrategyManager strategyManager = IStrategyManager(lrtConfig.getContract(LRTConstants.EIGEN_STRATEGY_MANAGER));
+
+        // For each asset
+        address[] memory assets = lrtConfig.getSupportedAssetList();
+        for (uint256 i = 0; i < assets.length;) {
+            // Get the EigenLayer strategy for the asset
+            // The WETH asset will point to the EigenLayer beaconChainETHStrategy
+            // 0xbeaC0eeEeeeeEEeEeEEEEeeEEeEeeeEeeEEBEaC0
```

Comment:
This comment appears wrong?

<img width="327" alt="image" src="https://github.com/oplabs/primestaked-eth/assets/837/27802735-6bc1-402c-9221-0b0699c6e344">

WETH returns a zero address from `lrtConfig.assetStrategy`. Fortunately ` strategyManager.stakerStrategyShares` will return a 0 for any bogus strategy, but still I think we should skip the rest of the loop body if `strategy` is address(0);

Reply:
Skipping will also match the behavior of `getAssetBalance`, which also skips checking the strategy if the asset is zero.

Reply:
yes, the comment was wrong. I think we should explicitly skip WETH to match the `getAssetBalance` logic.
```Solidity
// Skip WETH as ETH restaked into EigenLayer is not yet supported by the NodeDelegator.
if (assets[i] == WETH) {
    continue;
}
```